examples/cpp/benchmark/PublisherApp.cpp

```diff
@@ -0,0 +1,545 @@
+                }
+            }
+            break;
+
+        default:
+            throw std::runtime_error("Type invalid");
+    }
+}
+
+void PublisherApp::run()
+{
+    uint16_t prevCount = 0;
+    while (!is_stopped() && !publish())
+    {
+        // Wait for period
+        std::unique_lock<std::mutex> initial_lock(mutex_);
+        auto check = cv_.wait_for(initial_lock, std::chrono::milliseconds(1), [&]()
+                        {
+                            return is_stopped();
+                        });
+        if (check)
+        {
+            return;
+        }
+    }
+    {
+        // Wait for period
+        std::unique_lock<std::mutex> wait_lock(mutex_);
+        auto check = cv_.wait_for(wait_lock, std::chrono::milliseconds(wait_), [&]()
+                        {
+                            return is_stopped();
+                        });
+        if (check)
+        {
+            return;
+        }
+        count = 0;
+    }
+
+    auto actualTime = std::chrono::steady_clock::now();
+    auto elapsed = std::chrono::duration_cast<std::chrono::milliseconds>(actualTime - startTime);
+    while (!is_stopped() && elapsed.count() < end_)
+    {
+        // Wait for period or stop event
+        std::unique_lock<std::mutex> periodic_lock(mutex_);
+        auto check = cv_.wait_for(periodic_lock, std::chrono::milliseconds(period_ms_), [&]()
+                        {
+                            return is_stopped();
+                        });
+        if (check)
+        {
+            return;
+        }
+        vSamples.push_back(static_cast<uint16_t>(count) - prevCount);
+        prevCount = static_cast<uint16_t>(count);
+        actualTime = std::chrono::steady_clock::now();
+        elapsed = std::chrono::duration_cast<std::chrono::milliseconds>(actualTime - startTime);
+    }
+    std::cout << "RESULTS after " << elapsed.count() << " milliseconds:" << std::endl;
+    switch (msg_size_)
+    {
+        case CLIParser::MsgSizeKind::NONE:
+            std::cout << "COUNT: " << benchmark_.index() << std::endl;
+            break;
+
+        case CLIParser::MsgSizeKind::SMALL:
+            std::cout << "COUNT: " << benchmark_small_.index() << std::endl;
+            break;
+
+        case CLIParser::MsgSizeKind::MEDIUM:
+            std::cout << "COUNT: " << benchmark_medium_.index() << std::endl;
+            break;
+
+        case CLIParser::MsgSizeKind::BIG:
+            std::cout << "COUNT: " << benchmark_big_.index() << std::endl;
+            break;
+
+        default:
+            throw std::runtime_error("Type invalid");
+    }
+    std::cout << "SAMPLES: ";
+
+    for (uint16_t i = 0; i < vSamples.size(); ++i)
+    {
+        std::cout << vSamples[i] << ",";
+    }
+    std::cout << std::endl;
+
+    // Wait in case a response is still nedded
+    std::unique_lock<std::mutex> final_lock(mutex_);
+    cv_.wait(final_lock, [&]()
+            {
+                return is_stopped() || (sent == 0);
+            });
```

Comment:
I would say we dont need this logic at the end, the test-time has already finished and I find no reason for waiting a final response.
In addition, there is a typo in the comment (`nedded`).
If this suggestion is accepted, I think we could we could get rid of the `sent` member